pkg/cmd/extension/example/cmd_add_test.go

```diff
@@ -15,243 +15,176 @@
+		},
+	}
 
-		dir, err = filepath.EvalSymlinks(dir)
-		Expect(err).ToNot(HaveOccurred())
+	for _, test := range tests {
+		test := test // pin! see https://github.com/kyoh86/scopelint for why
 
-		err = os.Chdir(dir)
-		Expect(err).ToNot(HaveOccurred())
+		t.Run(test.name, func(t *testing.T) {
+			// Run "getenvoy extension examples add" with the args we are testing
+			c, stdout, stderr := cmd.NewRootCommand()
+			c.SetArgs(append([]string{"extension", "examples", "add"}, test.args...))
+			err := cmdutil.Execute(c)
+			require.EqualError(t, err, test.expectedErr, `expected an error running [%v]`, c)
 
-		return dir
+			// Verify the command failed with the expected error
+			require.Empty(t, stdout.String(), `expected no stdout running [%v]`, c)
+			expectedStderr := fmt.Sprintf("Error: %s\n\nRun 'getenvoy extension examples add --help' for usage.\n", test.expectedErr)
+			require.Equal(t, expectedStderr, stderr.String(), `unexpected stderr running [%v]`, c)
+		})
+	}
+}
+
+func TestGetEnvoyExtensionExamplesAddFailsOutsideWorkspaceDirectory(t *testing.T) {
+	// Change to a non-workspace dir
+	dir, revertWd := RequireChDir(t, relativeRustWorkspaceDirWithOneExample+"/..")
+	defer revertWd()
+
+	// Run "getenvoy extension examples add"
+	c, stdout, stderr := cmd.NewRootCommand()
+	c.SetArgs([]string{"extension", "examples", "add"})
+	err := cmdutil.Execute(c)
+
+	// Verify the command failed with the expected error
+	expectedErr := "there is no extension directory at or above: " + dir
+	require.EqualError(t, err, expectedErr, `expected an error running [%v]`, c)
+	require.Empty(t, stdout.String(), `expected no stdout running [%v]`, c)
+	expectedStderr := fmt.Sprintf("Error: %s\n\nRun 'getenvoy extension examples add --help' for usage.\n", expectedErr)
+	require.Equal(t, expectedStderr, stderr.String(), `unexpected stderr running [%v]`, c)
+}
+
+func TestGetEnvoyExtensionExamplesAddFailsWhenExampleExists(t *testing.T) {
+	_, revertWd := RequireChDir(t, relativeRustWorkspaceDirWithOneExample)
+	defer revertWd()
+
+	// Run "getenvoy extension examples add"
+	c, stdout, stderr := cmd.NewRootCommand()
+	c.SetArgs([]string{"extension", "examples", "add"})
+	err := cmdutil.Execute(c)
+
+	// Verify the command failed with the expected error
+	expectedErr := `example setup "default" already exists`
+	require.EqualError(t, err, expectedErr, `expected an error running [%v]`, c)
+	require.Empty(t, stdout.String(), `expected no stdout running [%v]`, c)
+	expectedStderr := fmt.Sprintf("Error: %s\n\nRun 'getenvoy extension examples add --help' for usage.\n", expectedErr)
+	require.Equal(t, expectedStderr, stderr.String(), `unexpected stderr running [%v]`, c)
+}
+
+func TestGetEnvoyExtensionExamplesAdd(t *testing.T) {
+	type testCase struct {
+		name                   string
+		templateWorkspace      string
+		flags                  []string
+		flagValues             []string
```

Comment:
can these be grouped together as `args []string` like others? 😄

Reply:
sure I'll make a sweep on this. I meant to convert them all..